pkg/cmd/server/origin/legacy.go

```diff
@@ -186,9 +189,38 @@ func LegacyStorage(storage map[schema.GroupVersion]map[string]rest.Storage) map[
 	for _, gvStorage := range storage {
 		for resource, s := range gvStorage {
 			if OriginLegacyResources.Has(resource) || OriginLegacySubresources.Has(resource) {
-				legacyStorage[resource] = s
+				// We want *some* our legacy resources to orphan by default instead of garbage collecting.
+				// Kube only did this for a select few resources which were controller managed and established links
+				// via a workload controller.  In openshift, these will all conform to registry.Store so we
+				// can actually wrap the "normal" storage here.
+				switch resource {
+				case "buildConfigs":
+					restStorage := *s.(*buildconfigetcd.REST)
```

Comment:
I think the shallow copy allows me change the deletestrategy but still share the core storage aspects like the cache.